nio_command_block.py

```diff
@@ -0,0 +1,173 @@
+from nio.metadata.properties import IntProperty, ListProperty, \
+    SelectProperty, ObjectProperty, PropertyHolder, StringProperty, \
+    TimeDeltaProperty, ExpressionProperty
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.common.signal.base import Signal
+from nio.modules.scheduler import Job
+from urllib.parse import urlencode
+import base64
+from .oauth2_mixin.oauth2 import OAuth2, OAuth2Exception
+import requests
+from enum import Enum
+
+
+class SecurityMethod(Enum):
+    NONE = 'none'
+    BASIC = 'basic'
+    OAUTH = 'oauth'
+
+
+class URLParameter(PropertyHolder):
+    prop_name = ExpressionProperty(title="Property Name")
+    prop_value = ExpressionProperty(title="Property Value")
+
+
+class BasicAuthCreds(PropertyHolder):
+    username = StringProperty(title='Username')
+    password = StringProperty(title='Password')
+
+
+@Discoverable(DiscoverableType.block)
+class NioCommand(Block, OAuth2):
```

Comment:
We want mixins to be listed first, mainly because of the way super calls and the MRO are processed

```python
class NioCommand(OAuth2, Block)
```